### PR TITLE
Use strings for ZIP codes, add validation

### LINKED ISSUE #7: Treat ZIP codes as strings
ZIP codes should be strings, not integers, as they can have leading 0s.

### PR DESCRIPTION
**Pull request by briandconnelly**
- Update type hints and doc strings for ZIP codes from `int` to `str`
- Add `validate_zip_code` function, which raises errors for invalid values
- Apply ZIP code validation to CLI passed arguments

closes #7

### CODE REVIEW

File: airnow/misc.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+
+def validate_zip_code(x):
```
Comment (j4c0bs) [inline airnow/misc.py]: Think this func will be used elsewhere? If its only validating args, maybe we should rename this file to something like `validate_args` and include `valid_iso_8601` from my upcoming PR. Also, we could raise `argparse.ArgumentTypeError` and stick w the whole proper argparse approach.

**Review by j4c0bs — APPROVED** [review]

File: airnow/misc.py (hunk 1 above)
Comment (briandconnelly) [reply airnow/misc.py]: I was thinking we might use it again for the wrapper functions (stuff like `conditions_zip`, `forecast_zip`, etc.), which is why I put it in a separate file and used the more generic error types. Definitely agree with renaming the file to something more sensible, though.

Comment (briandconnelly) [reply airnow/misc.py]: Moved to `validation.py`. I think maybe moving the date validation logic into there would be nice too. (side note: a whole package dedicated to validating stuff could be HUGE).

Comment (j4c0bs) [reply airnow/misc.py]: Great idea. I'll rebase and move `valid_iso_8601`.

**Pull request merged**